src/code/z_parameter.c

```diff
@@ -1733,12 +1733,16 @@ u8 Item_Give(PlayState* play, u8 item) {
         Health_ChangeBy(play, 0x10);
         return item;
     } else if (item == ITEM_MAGIC_JAR_SMALL) {
+#if OOT_VERSION < PAL_1_0
+        Magic_Fill(play);
+#else
         if (gSaveContext.magicState != MAGIC_STATE_ADD) {
             // This function is only used to store the magicState.
             // Setting the state to FILL gets immediately overwritten in Magic_RequestChange.
             // I.e. magic is added not filled
             Magic_Fill(play);
```

Comment:
the comment applies to both versions, maybe move it to above the #if and change "this function" to "magic_fill"
(also in the jar_big case)

Reply:
The magic system is extremely bad, here Magic_Fill is used to store the current magicState in prevMagicState so that it can be restored later. The comment could be clearer perhaps but yeah whatever